core/arch/arm/kernel/kern.ld.S

```diff
@@ -379,7 +374,11 @@ SECTIONS
 
 }
 
-#if defined(CFG_TEE_COHERENT_START)
-ASSERT((__coherent_start & (4096 - 1)) == 0, "Coherent RAM start is not 4Kbyte aligned")
-ASSERT((__coherent_end & (4096 - 1)) == 0, "Coherent RAM end is not 4Kbyte aligned")
+#if CFG_TEE_COHERENT_SIZE
+ASSERT(!(__coherent_start & (4096 - 1)),
```

Comment:
I don't see the problem with wrapping this inside the `DECLARE_COHERENT_RAM_SECTION` macros as:
```
                ASSERT(!(__coherent_start & (4096 - 1)),    \
                       "Coherent memory start alignment");     \
```